### PR TITLE
Fix : Local development failing due to changed function definition

### LINKED ISSUE #773: [Bug]: Local development record, replay and start fail
### Describe the bug

So the current local development setup is buggy due to recent code changes. The helper function definitions were changed but the usage was not updates.
Ex: functions for recording start, replay etc

### To Reproduce

Any record start will fail

### PR DESCRIPTION
**Pull request by shashank40**
**What kind of change does this PR introduce?**
fixes #773 

**Summary**

The current local development setup is buggy due to recent code changes. The helper function definitions were changed but the usage was not updates.
Ex: functions for recording start, replay etc

**Checklist**
* [ ] My code follows the style guidelines of OpenAdapt
* [ ] I have performed a self-review of my code
* [ ] If applicable, I have added tests to prove my fix is functional/effective
* [ ] I have linted my code locally prior to submission
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [ ] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

**Other information**

### CODE REVIEW

File: openadapt/start.py [hunk 1]
```diff
@@ -24,7 +25,7 @@ def main() -> None:
         subprocess.run(["git", "pull", "-q"])
         logger.info("Updated the OpenAdapt App")
 
-    run_app()  # start gui
+    start()  # start gui
```
Comment (abrichr) [inline openadapt/start.py:28]: @shashank40 thank you for putting this together!

@KIRA009 do you know whether we are using this file `openadapt/start.py` anywhere? Git blame shows last update was about a year go

File: openadapt/plotting.py [hunk 2]
```diff
@@ -393,11 +395,11 @@ def plot_performance(
 
         ax.legend(all_handles, all_labels)
 
-    ax.set_title(f"{recording_timestamp=}")
+    ax.set_title(f"{recording.timestamp=}")
 
     # TODO: add PROC_WRITE_BY_EVENT_TYPE
     if save_file:
-        fname_parts = ["performance", str(recording_timestamp)]
+        fname_parts = ["performance", str(recording.timestamp)]
```
Comment (abrichr) [inline openadapt/plotting.py:402]: This has been fixed in https://github.com/OpenAdaptAI/OpenAdapt/pull/774

File: openadapt/strategies/vanilla.py [hunk 3]
```diff
@@ -35,20 +35,20 @@ class VanillaReplayStrategy(strategies.base.BaseReplayStrategy):
     def __init__(
         self,
         recording: models.Recording,
-        replay_instructions: str = "",
+        instructions: str = "",
```
Comment (abrichr) [inline openadapt/strategies/vanilla.py:38]: @shashank40 if you remove the rest of this PR and only include the changes to this file I will be happy to merge it 👍

File: openadapt/strategies/vanilla.py [hunk 4]
```diff
@@ -195,7 +195,7 @@ def generate_action_event(
         current_window=current_window_dict,
         recorded_actions=recorded_action_dicts,
         replayed_actions=replayed_action_dicts,
-        replay_instructions=replay_instructions,
+        instructions=instructions,
```
Comment (abrichr) [inline openadapt/strategies/vanilla.py:198]: I believe this needs to stay as "replay_instructions", as per the .j2 file

File: openadapt/start.py (hunk 1 above)
Comment (shashank40) [reply openadapt/start.py:28]: @abrichr @KIRA009 as my app was not working after download, i used this script to spin up the app.

`python -m openadapter.start` runs a client for us and that is what i used this for

File: openadapt/plotting.py (hunk 2 above)
Comment (shashank40) [reply openadapt/plotting.py]: Yess @abrichr . i took the latest rebase and removed it

File: openadapt/start.py (hunk 1 above)
Comment (abrichr) [reply openadapt/start.py:28]: @shashank40 please feel free to move this file to `openadapt/deprecated`, and remove any mention of it in any documentation that you encountered 🙏

Comment (abrichr) [reply openadapt/start.py:28]: https://github.com/OpenAdaptAI/OpenAdapt/issues/777

File: openadapt/strategies/vanilla.py (hunk 4 above)
Comment (shashank40) [reply openadapt/strategies/vanilla.py:198]: But replay fails.
If we use the replay command with `--instructions`, it fails with the error, undefined argument instructions.

We can do 2 things
either the command we use has`--replay_instructions` as argument OR vanilla.py changes replay_instructions -> instructions.

I would change it to whatever you might prefer

Comment (abrichr) [reply openadapt/strategies/vanilla.py:198]: To clarify:

this line should read `replay_instructions=instructions`

Comment (shashank40) [reply openadapt/strategies/vanilla.py:198]: Oh yess, it would, missed the line number. You are correct

File: openadapt/start.py (hunk 1 above)
Comment (shashank40) [reply openadapt/start.py:28]: @KIRA009 
I meant the compiled app as i commented here.
https://github.com/OpenAdaptAI/OpenAdapt/pull/771#discussion_r1644517094

File: openadapt/strategies/vanilla.py (hunk 4 above)
Comment (shashank40) [reply openadapt/strategies/vanilla.py]: @abrichr updated the PR 🙏🏻

**Pull request closed without merging**